Framework/Muon/inc/MantidMuon/PSIBackgroundSubtraction.h

```diff
@@ -0,0 +1,44 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#pragma once
+
+#include "MantidAPI/Algorithm.h"
+#include "MantidMuon/DllConfig.h"
+
+namespace Mantid {
+namespace Muon {
+
+class MANTID_MUON_DLL PSIBackgroundSubtraction : public API::Algorithm {
```

Comment:
Am I correct in saying that this just runs fit and then a subtraction? I think this might be a good candidate for promoting to a so-called `WorkflowAlgorithm`. These inherit from `API::DataProcessorAlgorithm` and the only requirement for them is that all they do is run algorithms themselves - they are basically fancy scripts.

They have the advantage that when generating history from them you can "unroll" and see what was done and it makes the process more transparent.  Is it worth changing to this type of algorithm?

Reply:
This sounds like a good idea, the added transparency would definitely be beneficial.

Reply:
This sounds like a good idea, the added transparency would definitely be beneficial.